src/domTree.js

```diff
@@ -235,21 +256,20 @@ class anchor extends HtmlDomContainer<HtmlDomNode> {
 }
 
 /**
- * This node represents a document fragment, which contains elements, but when
- * placed into the DOM doesn't have any representation itself. Thus, it only
- * contains children and doesn't have any HTML properties. It also keeps track
- * of a height, depth, and maxFontSize.
+ * HTML version of the documentFragment that adds otherwise unused fields and
+ * methods to implement HtmlDomNode.
  */
-class documentFragment implements HtmlDomNode {
-    children: HtmlDomNode[];
-    classes: string[];         // Never used; needed for satisfying interface.
+export class documentFragment extends tree.documentFragment<HtmlDomNode>
+    implements HtmlDomNode {
+    // Never used; needed for satisfying interface.
+    classes: string[];
     height: number;
     depth: number;
     maxFontSize: number;
     style: CssStyle;          // Never used; needed for satisfying interface.
```

Comment:
Could you have a look at questions raised at https://github.com/Khan/KaTeX/issues/1439#issuecomment-399452762 and https://github.com/Khan/KaTeX/pull/1471#issuecomment-404193446? I've tried to make `classes` and `style` optional and/or separate `documentFragement` from `HtmlDomNode`, but it seems to be closely intertwined.

Reply:
I've coalesced `documentFragment` into a single class that implements both `HtmlDomNode` and `MathDomNode` interfaces. It required a `$FlowFixMe`.

I also ran into additional cases of this uninformative error that's most certainly a bug in flow:
```
Error: "documentFragment [1] is incompatible with MathNode [2]."
```
I've got 4 instances of it (one was "incompatible with symbolNode") that I had to suppress with `$FlowFixMe`. I'd appreciate any insights, or we could let it be.